File: javascript/leetcode/226. Invert Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
  if (!root) return root;
  if (!root.right && !root.left) return root;

  // swap children
  let temp = root.right;
  root.right = root.left;
  root.left = temp;

  invertTree(root.right);
  invertTree(root.left);

  return root;
};

File: c/leetcode/226. Invert Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

struct TreeNode *invertTree(struct TreeNode *root)
{
  if (root && (root->right || root->left))
  {
    struct TreeNode *tmp = root->right;
    root->right = root->left;
    root->left = tmp;

    invertTree(root->right);
    invertTree(root->left);
  }

  return root;
}
